resources/prosody-plugins/mod_muc_password_check.lua

```diff
@@ -0,0 +1,218 @@
+
+--- Verifies room name, domain name with the values in the token
+-- @param token the token we received
+-- @param room_address the full room address jid
+-- @return true if values are ok or false otherwise
+function verify_token(token, room_address)
+    if not enableTokenVerification then
+        return true;
+    end
+
+    -- if enableTokenVerification is enabled and we do not have token
+    -- stop here, cause the main virtual host can have guest access enabled
+    -- (allowEmptyToken = true) and we will allow access to rooms info without
+    -- a token
+    if token == nil then
+        module:log("warn", "no token provided");
+        return false;
+    end
+
+    local session = {};
+    session.auth_token = token;
+    local verified, reason, msg = token_util:process_and_verify_token(session);
+    if not verified then
+        module:log("warn", "not a valid token %s %s", tostring(reason), tostring(msg));
+        return false;
+    end
+
+    -- accepting server tokens, which currently do not specify room permissions, so skip this check
+    -- if not token_util:verify_room(session, room_address) then
+    --     log("warn", "Token %s not allowed to join: %s",
+    --         tostring(token), tostring(room_address));
+    --     return false;
+    -- end
+
+    return true;
+end
+
+function handle_validate_room_password (event)
+    local request = event.request;
+    module:log("info","Request for room password validate: reqid %s", request.headers["request_id"])
+
+    if request.headers.content_type ~= json_content_type
+            or (not request.body or #request.body == 0) then
+        return 400;
+    end
+
+    local params = json.decode(event.request.body);
+    if not params then
+        return 400;
+    end
+
+    local conference = params["conferenceFullName"];
+    local passcode = params["passcode"];
+
+    if (not conference or not passcode) then
+        return { status_code = 400 };
+    end
+
+    local room_address = room_jid_match_rewrite(conference)
+
+    -- verify access
+    local token = event.request.headers["authorization"]
+    if not token then
+        return { status_code = 403 };
+    end
+
+    if starts_with(token,'Bearer ') then
+        token = token:sub(8,#token)
+    end
+
+    --    module:log("debug","incoming token %s",token)
+    if not verify_token(token, room_address) then
+        return { status_code = 403 };
+    end
+
+    local room = get_room_from_jid(room_address);
+
+    response_data = {};
+    local PUT_response = {
+        headers = { content_type = "application/json"; };
+    };
+    if room then
+        response_data["valid"] = room:get_password() == passcode;
+    else
+        response_data["valid"] = false;
+
+        -- no room found but everything else was valid
+        PUT_response.status_code = 404;
+    end
+
+    PUT_response.body = json.encode(response_data);
+    module:log("info","Sending response for room password validate: %s", inspect(PUT_response));
```

Comment:
shouldn't this be debug level?

Reply:
I'm changing it ... but I'm hesitating ... maybe we should leave a trace in the logs for a check ...